mokapot/streaming.py

```diff
@@ -0,0 +1,285 @@
+from typing import Generator, Callable, Iterator, Type
```

Comment:
Would you mind adding comments/dosctrings exmplaining the intended use case for each of the classes?

Reply:
@ezander